File: tests/testA.py
import unittest
import pylua

lua_module = pylua.LuaLoad("testA.lua")

class TestBasic(unittest.TestCase):
    def test_simple_types(self):
        self.assertEqual(lua_module.string, "Lua")
        self.assertEqual(lua_module.int, 121)
        self.assertFalse(lua_module.boolean)

    @unittest.skip("Iteration not implemented")
    def test_complex_types(self):
        for i, j in lua_module.list1:
            self.assertEqual(i + 1, j)
        
        for i, j in lua_module.list2:
            self.assertEqual(i, j)

    def test_function_calls(self):
        self.assertEqual(lua_module.return121(), 121)
        lua_module.get_values(121, 1.5, False, "Python")


if __name__ == "__main__":
    unittest.main()

File: tests/testA.lua
lu = require('luaunit')


string = "Lua"
int = 121
boolean = false
list1 = {0, 1, 2, 3, 4, 5}
list2 = {1, 2, 3, 4, 5}


function return121()
    return 121
end

function get_values(a, b, c, d)
    lu.assertEquals(a, 121)
    lu.assertEquals(b, 1.5)
    lu.assertEquals(c, false)
    lu.assertEquals(d, "Python")
end

Rectangle = {length = 0, breadth = 0}

function Rectangle:new(length, breadth)
    o = {}
    setmetatable(o, self)
    self.__index = self
    o.length = length
    o.breadth = breadth
    return o
end

function Rectangle:__tostring()
    return "Rectangle(length = " .. tostring(self.length) .. ", Breadth = " .. tostring(self.breadth) .. ")"
end

function Rectangle:get_area()
    self.area = self.length * self.breadth
    return self.area
end
